### PR TITLE
The update dialog doesn't prompt again when clicked "Ask me in a day" and toggled distribute state off to on

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
## Description
1. Install the second latest Unity app from install portal.
2. Launch app, it prompt "App update available" dialog.
3. Click "Ask me in a day".
4. Then toggle distribute state off to on.
5. Observe whether the update dialog prompt.

## Related PRs or issues
[AB#65075](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/65075)

### CODE REVIEW

File: Assets/Puppet/PuppetAppCenter.cs [hunk 1]
```diff
@@ -114,7 +116,8 @@ private void Awake()
         AppSecretCached = PlayerPrefs.GetString(AppSecretKey, null);
         LogUrlCached = PlayerPrefs.GetString(LogUrlKey, null);
         MaxSizeCached = PlayerPrefs.GetString(MaxStorageSizeKey, null);
-        StartupTypeCached = PlayerPrefs.GetInt(StartupModeKey, (int) Microsoft.AppCenter.Unity.StartupType.Both);
+        StartupTypeCached = PlayerPrefs.GetInt(StartupModeKey, (int)Microsoft.AppCenter.Unity.StartupType.Both);
+        CustomDialog.isOn = PlayerPrefs.GetInt(FlagCustomDialog, 0) == 1 ? true : false;
```
Comment (owlstronaut) [inline Assets/Puppet/PuppetAppCenter.cs]:
```suggestion
        CustomDialog.isOn = PlayerPrefs.GetInt(FlagCustomDialog, 0) == 1;
```

File: AppCenterDemoApp/Assets/Puppet/PuppetAppCenter.cs [hunk 2]
```diff
@@ -115,6 +117,7 @@ private void Awake()
         LogUrlCached = PlayerPrefs.GetString(LogUrlKey, null);
         MaxSizeCached = PlayerPrefs.GetString(MaxStorageSizeKey, null);
         StartupTypeCached = PlayerPrefs.GetInt(StartupModeKey, (int) Microsoft.AppCenter.Unity.StartupType.Both);
+        CustomDialog.isOn = PlayerPrefs.GetInt(FlagCustomDialog, 0) == 1 ? true : false;
```
Comment (owlstronaut) [inline AppCenterDemoApp/Assets/Puppet/PuppetAppCenter.cs]:
```suggestion
        CustomDialog.isOn = PlayerPrefs.GetInt(FlagCustomDialog, 0) == 1;
```

**Review by owlstronaut — CHANGES_REQUESTED** [review]

File: AppCenterDemoApp/Assets/Puppet/PuppetAppCenter.cs [hunk 3]
```diff
@@ -208,6 +208,16 @@ private IEnumerator SetEnabledCoroutine(bool enabled)
         DistributeEnabled.isOn = isDistributeEnabled.Result;
     }
 
+    public void SetCustomDialog(bool enable)
+    {
+        int isEnable = enable ? 1 : 0;
+        PlayerPrefs.SetInt(FlagCustomDialog, isEnable);
+        PlayerPrefs.Save();
+#if UNITY_ANDROID && !UNITY_EDITOR
```
Comment (annakocheshkova) [inline AppCenterDemoApp/Assets/Puppet/PuppetAppCenter.cs]: You can remove this. It's only if you need to access the preferences from native code.

File: AppCenterDemoApp/Assets/Puppet/PuppetReleaseHandler.cs [hunk 4]
```diff
@@ -3,25 +3,32 @@
...
-        Distribute.ReleaseAvailable = details =>
+        Distribute.ReleaseAvailable = OnReleaseAvailable;
+    }
+
+    bool OnReleaseAvailable(ReleaseDetails details)
+    {
+        lock (_releaseLock)
         {
-            lock (_releaseLock)
+            if (PlayerPrefs.GetInt(PuppetAppCenter.FlagCustomDialog, 0) == 1)
```
Comment (annakocheshkova) [inline AppCenterDemoApp/Assets/Puppet/PuppetReleaseHandler.cs:23]: That's not enough. You need to add the check inside `Update` too.

**Review by annakocheshkova — CHANGES_REQUESTED** [review]

**Review by annakocheshkova — DISMISSED** [review]

**Review by Jamminroot — CHANGES_REQUESTED** [review]
.

**Review by Jamminroot — DISMISSED** [review]

File: Assets/AppCenter/Plugins/AppCenterSDK/Distribute/Shared/UpdateAction.cs [hunk 5]
```diff
@@ -1,21 +1,35 @@
...
+    /// <summary>
+    /// User update action.
+    /// </summary>
+    public enum UpdateAction
     {
-        /// <summary>
-        /// Action to trigger the download of the release.
-        /// </summary>
-        Update,
+#if UNITY_ANDROID
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/AppCenterSDK/Distribute/Shared/UpdateAction.cs]: It fixes the SDK. Please update changelog

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

**Review by owlstronaut — DISMISSED** [review]

File: AppCenterDemoApp/Assets/Puppet/PuppetReleaseHandler.cs [hunk 6]
```diff
@@ -3,25 +3,32 @@
 
 using Microsoft.AppCenter.Unity.Distribute;
 using UnityEngine;
+using System.Threading;
+using AOT;
```
Comment (jaeklim) [inline AppCenterDemoApp/Assets/Puppet/PuppetReleaseHandler.cs]: Order?

File: Assets/AppCenter/Plugins/AppCenterSDK/Distribute/Shared/UpdateAction.cs (hunk 5 above)
Comment (jaeklim) [reply Assets/AppCenter/Plugins/AppCenterSDK/Distribute/Shared/UpdateAction.cs]: What does it fix?

File: Assets/Puppet/PuppetReleaseHandler.cs [hunk 7]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 
+using System.Threading;
+using AOT;
```
Comment (jaeklim) [inline Assets/Puppet/PuppetReleaseHandler.cs]: Order

**Review by jaeklim — CHANGES_REQUESTED** [review]
What does *State off* mean? Does it mean disabling Distribute?

File: Assets/AppCenter/Plugins/AppCenterSDK/Distribute/Shared/UpdateAction.cs (hunk 5 above)
Comment (MatkovIvan) [reply Assets/AppCenter/Plugins/AppCenterSDK/Distribute/Shared/UpdateAction.cs]: @jaelim-ms it changes constants for android (Did it work before?)

Well, actually I would keep this public part as is and fix the Android binding if required

**Review by dtrabo — DISMISSED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,13 @@
 # App Center SDK for Unity Change Log
 
+## Release 2.2.1 (Under development)
+
+**Android**
+
+* **[Bug fix]** Fixed handler the update actions in custom update dialog in Distributed module.
```
Comment (annakocheshkova) [inline CHANGELOG.md]:
```suggestion
* **[Bug fix]** Fixed handling the update actions in custom update dialog in Distributed module.
```

**Review by annakocheshkova — DISMISSED** [review]

**Review by annakocheshkova — APPROVED** [review]

**Review by MatkovIvan — APPROVED** [review]

**Review by jaeklim — APPROVED** [review]

**Pull request merged**